tests/cluster/tests/includes/init-tests.tcl

```diff
@@ -32,6 +32,17 @@ test "Cluster nodes hard reset" {
         } else {
             set node_timeout 3000
         }
+
+        # Wait until slave is synced. Otherwise, it may reply -LOADING
+        # for any commands below.
+        if {[RI $id role] eq {slave}} {
+            wait_for_condition 50 1000 {
+                [RI $id master_link_status] eq {up}
+            } else {
+                fail "Slave were not able to sync."
+            }
+        }
```

Comment:
i wonder if the fact we had so many issues in the test suite, it an indication that we may be breaking some apps?

Reply:
yeah, before we would have replied with LOADING only in case of long rdb parsing phase.
scenarios with long flushing, and short rdb loading phase could be sensitive to this change.

in this specific case, it's a test infra code that calls FLUSHALL, and who knows in which scenarios it is executed. so this change is ok..
maybe the other one mentioned above (with `CONFIG SET appendonly`) is different and should be handled.

Reply:
Pushed the change to allow `appendonly` config change during loading.